invenio_rdm_records/services/services.py

```diff
@@ -86,3 +88,38 @@ def scan_expired_embargos(self, identity):
             f"AND access.embargo.until:[* TO {today}]"
 
         return self.scan(identity=identity, q=embargoed_q)
+
+    def search_community_records(self, identity, community, params=None,
+                                 es_preference=None, **kwargs):
+        """Search for drafts records matching the querystring."""
+        community_uuid = community.to_dict()['uuid']
+        self.require_permission(identity, "read")
+
+        # Prepare and execute the search
+        params = params or {}
+
+        search_result = self._search(
+            "search",
+            identity,
+            params,
+            es_preference,
+            record_cls=self.record_cls,
+            search_opts=self.config.search,
+            extra_filter=Q(
+                "term", **{"parent.communities.default": str(community_uuid)}
```

Comment:
I am not sure this is enough. A record can be in the default community, or in one of the `ids` list.
Please add tests for both cases.